api/v1alpha1/flowcollector_types.go

```diff
@@ -622,6 +611,31 @@ type ClientTLS struct {
 	UserCert CertificateReference `json:"userCert,omitempty"`
 }
 
+// DebugConfig allows tweaking some aspects of the internal configuration of the agent and FLP.
+// They are aimed exclusively for debugging. Users setting these values do it at their own risk.
+type DebugConfig struct {
+	//+kubebuilder:validation:Enum=IfNotPresent;Always;Never
+	//+kubebuilder:default:=IfNotPresent
+	// imagePullPolicy is the Kubernetes pull policy for the component image.
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
```

Comment:
Ok. I moved it here because it looked a bit strange that the images are not here anymore but the ImagePullPolicy is actually here.

Also because I was thinking on, after GA, we could restore the "image" fields under the debug section so they would override the CSV `relatedImages` entry if they were defined here. So we wouldn't need to patch the CSV or deployment during development.